seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/execution/FlinkRuntimeEnvironment.java

```diff
@@ -186,158 +82,15 @@ private void createStreamTableEnvironment() {
-                String uri = config.getString(ConfigKeyName.CHECKPOINT_DATA_URI);
-                StateBackend fsStateBackend = new FsStateBackend(uri);
-                if (EnvironmentUtil.hasPathAndWaring(config, ConfigKeyName.STATE_BACKEND)) {
-                    String stateBackend = config.getString(ConfigKeyName.STATE_BACKEND);
-                    if ("rocksdb".equalsIgnoreCase(stateBackend)) {
-                        StateBackend rocksDBStateBackend =
-                                new RocksDBStateBackend(fsStateBackend, TernaryBoolean.TRUE);
-                        environment.setStateBackend(rocksDBStateBackend);
-                    }
-                } else {
-                    environment.setStateBackend(fsStateBackend);
-                }
-            }
-
-            if (EnvironmentUtil.hasPathAndWaring(
-                    config, ConfigKeyName.MAX_CONCURRENT_CHECKPOINTS)) {
-                int max = config.getInt(ConfigKeyName.MAX_CONCURRENT_CHECKPOINTS);
-                checkpointConfig.setMaxConcurrentCheckpoints(max);
-            }
-
-            if (EnvironmentUtil.hasPathAndWaring(config, ConfigKeyName.CHECKPOINT_CLEANUP_MODE)) {
-                boolean cleanup = config.getBoolean(ConfigKeyName.CHECKPOINT_CLEANUP_MODE);
-                if (cleanup) {
-                    checkpointConfig.enableExternalizedCheckpoints(
-                            CheckpointConfig.ExternalizedCheckpointCleanup.DELETE_ON_CANCELLATION);
-                } else {
-                    checkpointConfig.enableExternalizedCheckpoints(
-                            CheckpointConfig.ExternalizedCheckpointCleanup.RETAIN_ON_CANCELLATION);
-                }
-            }
-
-            if (EnvironmentUtil.hasPathAndWaring(
-                    config, ConfigKeyName.MIN_PAUSE_BETWEEN_CHECKPOINTS)) {
-                long minPause = config.getLong(ConfigKeyName.MIN_PAUSE_BETWEEN_CHECKPOINTS);
-                checkpointConfig.setMinPauseBetweenCheckpoints(minPause);
-            }
-
-            if (EnvironmentUtil.hasPathAndWaring(
-                    config, ConfigKeyName.FAIL_ON_CHECKPOINTING_ERRORS)) {
-                int failNum = config.getInt(ConfigKeyName.FAIL_ON_CHECKPOINTING_ERRORS);
-                checkpointConfig.setTolerableCheckpointFailureNumber(failNum);
-            }
-        }
-    }
-
-    public void registerResultTable(Config config, DataStream<Row> dataStream, String name) {
-        StreamTableEnvironment tableEnvironment = this.getStreamTableEnvironment();
-        if (!TableUtil.tableExists(tableEnvironment, name)) {
-            tableEnvironment.createTemporaryView(
-                    name, tableEnvironment.fromChangelogStream(dataStream));
+    protected void setCheckpoint() {
+        super.setCheckpoint();
+        CheckpointConfig checkpointConfig = environment.getCheckpointConfig();
+        if (EnvironmentUtil.hasPathAndWaring(config, EnvCommonOptions.CHECKPOINT_TIMEOUT.key())) {
```

Comment:
it should be same with another runtime environment. I believe it was a mistake.

Reply:
> it should be same with another runtime environment. I believe it was a mistake.

[https://github.com/apache/seatunnel/pull/7040#discussion_r1669436408](https://github.com/apache/seatunnel/pull/7040#discussion_r1669436408)
Can you help me confirm this? If there is any problem, I will modify them to be the same.

Reply:
Yes. Please use `if (config.hasPath(EnvCommonOptions.CHECKPOINT_TIMEOUT.key())) {`.

Reply:
> Yes. Please use `if (config.hasPath(EnvCommonOptions.CHECKPOINT_TIMEOUT.key())) {`.

@Hisoka-X modify completed